ir/src/integrity_stmts/mod.rs

```diff
@@ -98,10 +98,12 @@ impl IntegrityStmts {
             IntegrityStmt::Constraint(constraint) => {
                 let expr = constraint.expr();
 
+                let mut row_offsets = Vec::new();
                 // add it to the integrity constraints graph and get its entry index.
                 let (trace_segment, root_index) = self.constraints_graph.insert_expr(
                     symbol_table,
                     expr,
+                    &mut row_offsets,
```

Comment:
I don't think that this should be added as a parameter. In the case of the `variable_roots` we have to pass it as a parameter, because we need to track them and modify the set of them across many constraints. In the case of the `row_offsets`, this is specific to each constraint, just like the `trace_segment` or `root_index`. Therefore, I think we should follow the same pattern as the `root_index` and have this be part of the return value of `insert_expr`